compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -2799,7 +2799,8 @@ public void visit(BLangExpressionStmt exprStmtNode, AnalyzerData data) {
         BType bType = typeChecker.checkExpr(expr, stmtEnv, data.prevEnvs, data.commonAnalyzerData);
         if (bType != symTable.nilType && bType != symTable.semanticError &&
                 expr.getKind() != NodeKind.FAIL &&
-                !types.isNeverTypeOrStructureTypeWithARequiredNeverMember(bType)) {
+                !types.isNeverTypeOrStructureTypeWithARequiredNeverMember(bType)
+                && !types.isAssignable(bType, symTable.nilType)) {
```

Comment:
We don't need `bType != symTable.nilType` separately with this? Let's make this the first check instead.